src/core/contourFeature.js

```diff
@@ -194,9 +198,55 @@ geo.contourFeature = function (arg) {
     if (gridW * gridH > data.length) {
       gridH = Math.floor(data.length) / gridW;
     }
+    /* If we are not using the position values (we are using x0, y0, dx, dy),
+     * and wrapLongitude is turned on, and the position spans 180 degrees,
+     * duplicate one or two columns of points at opposite ends of the map. */
+    usePos = (x0 === null || x0 === undefined || y0 === null ||
+        y0 === undefined || !dx || !dy);
+    if (!usePos && result.wrapLongitude && (x0 < -180 || x0 > 180 ||
+        x0 + dx * (gridW - 1) < -180 || x0 + dx * (gridW - 1) > 180) &&
```

Comment:
Agreed, what is done here is quite specific to the contouring algorithm.  I hadn't thought about just repeating features every 360 degrees, but that is a good idea.  I was already adding in my branch a concept of periodicity to the tile layer, but I can expose that through the map so that it can be used here.